torch_geometric/nn/encoding.py

```diff
@@ -94,3 +96,182 @@ def forward(self, x: Tensor) -> Tensor:
+            a linear layer encoding the output of :class:`TemporalEncoding`
+            concatenated with ``edge_attr``.
+        time_channels (int): dims to encode each timestamp into with
+            :class:`TemporalEncoding`.
+        out_channels (int): Size of each output sample.
+        is_sorted (bool, optional): If set to :obj:`True`, assumes that
+            :obj:`edge_index` is sorted by column. This avoids internal
+            re-sorting of the data and can improve runtime and memory
+            efficiency. (default: :obj:`False`)
+
+    Example:
+
+        >>> # GraphMixer paper uses the following args for GDELTLite dataset
+        >>> link_encoder = LinkEncoding(
+        ...     K=30,
+        ...     in_channels=186,
+        ...     hidden_channels=100,
+        ...     out_channels=100,
+        ...     time_channels=100,
+        ... )
+
+    """
+    def __init__(
+        self,
+        K: int,
+        in_channels: int,
+        hidden_channels: int,
+        out_channels: int,
+        time_channels: int,
+        is_sorted: bool = False,
+    ) -> None:
+        super().__init__()
+        self.K = K
+        self.in_channels = in_channels
+        self.hidden_channels = hidden_channels
+        self.out_channels = out_channels
+        self.time_channels = time_channels
+        self.is_sorted = is_sorted
+
+        # teomporal encoder
+        self.temporal_encoder = TemporalEncoding(time_channels)
+        self.temporal_encoder_head = torch.nn.Linear(
+            time_channels + in_channels,
+            hidden_channels,
+        )
+
+        # temporal information summariser
+        self.mlp_mixer = _MLPMixer(
+            num_tokens=K,
+            in_channels=hidden_channels,
+            out_channels=out_channels,
+        )
+
+    def forward(
+        self,
+        edge_attr: Tensor,
+        edge_time: Tensor,
+        edge_index: Tensor,
+    ) -> Tensor:
+        """
+        Args:
+            edge_attr (torch.Tensor): ``[num_edges, in_channels]``
+            edge_time (torch.Tensor): ``[num_edges,]``
+            edge_index (torch.Tensor): ``[2, num_edges]``
+
+        Returns:
+            A tensor of size ``[num_nodes, out_channels]``
+        """
+        time_info = self.temporal_encoder(edge_time)
+        edge_attr_time = torch.cat((time_info, edge_attr), dim=1)
+        edge_attr_time = self.temporal_encoder_head(edge_attr_time)
+
+        # `to_dense_batch` assumes sorted inputs
+        if not self.is_sorted:
+            edge_index[1], indices = index_sort(edge_index[1])
+            edge_attr_time = edge_attr_time[indices]
+
+        # zero-pad each node's edges:
+        # [num_edges, hidden_channels] -> [num_nodes*K, hidden_channels]
+        edge_attr_time, _ = to_dense_batch(
+            edge_attr_time,
+            edge_index[1],
+            max_num_nodes=self.K,
+        )
+        return self.mlp_mixer(
+            edge_attr_time.view(-1, self.K, self.hidden_channels))
+
```

Comment:
If the `edge_index` did get sorted then they need to be put back in their original order.
```suggestion
        out= self.mlp_mixer(
            edge_attr_time.view(-1, self.K, self.hidden_channels))
        if indices is not None:
            out_unsorted = torch.empty_like(out_attr)
            out_unsorted[indices] = out
            out = out_unsorted
       return out

```

Reply:
@wsad1 Thank you for actively reviewing my PR!

One thing that's not obvious is that the input and output of `forward()` don't share the same shape in any dim because edges of each node are squashed into a single vector:
```
[num_edges, in_channels]   # edge_attr
[num_nodes, out_channels]  # out
```
and I'm not sure what the unsorted tensor would look like.

One reasonable way I can think of is to keep the output sorted and just document that the output tensor is always sorted by the node index.

What do you think?

Reply:
Apologies at @akihironitta , you don't need to make any changes. I forgot that the output if shape `num_nodes, out_channels`.